flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpClient.java

```diff
@@ -60,7 +60,7 @@ public DatadogHttpClient(
             String dgApiKey,
             String dgProxyHost,
             int dgProxyPort,
-            DataCenter dataCenter,
+            String dataCenterUrl,
```

Comment:
I think we should add a new constructor and deprecate the old one , for backwards compatibility.
These methods should all be tagged with @publicEvolving or @Internal as appropriate.
We should add @deprecate to DataCenter also.